lib/bundler/definition.rb

```diff
@@ -94,7 +94,7 @@ def initialize(lockfile, dependencies, sources, unlock, ruby_version = nil, opti
       end
       @unlocking ||= @unlock[:ruby] || (!@locked_ruby_version ^ !@ruby_version)
 
-      @gem_version_promoter = GemVersionPromoter.new(@locked_specs, @unlock[:gems])
+      create_gem_version_promoter
```

Comment:
Do you mean inline #create_gem_version_promoter? constructor already feels rather fat to me - I wanted to encapsulate that in its own method with the additional logic to re-capture the locked_specs. I tried to find a way to re-use the @locked_specs or locked local var rather than re-executing, but it didn't seem clean to me with the various conditionals.

But if that's not what you mean, I'm not understanding your comment ... ?